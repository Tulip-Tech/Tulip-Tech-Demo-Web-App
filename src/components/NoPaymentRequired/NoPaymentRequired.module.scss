@use 'src/styles/variables';
@use 'src/styles/theme';

.noPaymentRequired {
  p {
    font-weight: var(--body-font-weight-bold);
    font-size: 20px;
    text-align: center;
  }
}
