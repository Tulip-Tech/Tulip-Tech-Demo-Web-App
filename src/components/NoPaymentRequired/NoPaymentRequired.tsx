import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './NoPaymentRequired.module.scss';

import Button from '#components/Button/Button';
import FormFeedback from '#components/FormFeedback/FormFeedback';

type Props = {
  onSubmit?: () => void;
  error?: string;
};

const NoPaymentRequired: React.FC<Props> = ({ onSubmit, error }) => {
  const { t } = useTranslation('account');

  return (
    <div className={styles.noPaymentRequired}>
      {error ? <FormFeedback variant="error">{error}</FormFeedback> : null}
      <p>{t('checkout.no_payment_needed')}</p>
      <Button label={t('checkout.continue')} variant="contained" color="primary" size="large" onClick={onSubmit} fullWidth />
    </div>
  );
};

export default NoPaymentRequired;
