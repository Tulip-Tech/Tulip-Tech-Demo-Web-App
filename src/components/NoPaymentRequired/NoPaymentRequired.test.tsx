import React from 'react';
import { render } from '@testing-library/react';

import NoPaymentRequired from './NoPaymentRequired';

describe('<NoPaymentRequired>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<NoPaymentRequired />);

    expect(container).toMatchSnapshot();
  });
});
