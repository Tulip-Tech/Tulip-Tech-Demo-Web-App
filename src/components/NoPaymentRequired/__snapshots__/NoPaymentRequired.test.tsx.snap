// Vitest Snapshot v1

exports[`<NoPaymentRequired> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_noPaymentRequired_b113e6"
  >
    <p>
      checkout.no_payment_needed
    </p>
    <button
      class="_button_f8f296 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
    >
      <span>
        checkout.continue
      </span>
    </button>
  </div>
</div>
`;
